nwb_conversion_tools/datainterfaces/cellexplorerdatainterface.py

```diff
@@ -0,0 +1,66 @@
+"""Authors: Cody Baker and Ben Dichter."""
+from pathlib import Path
+
+import spikeextractors as se
+from scipy.io import loadmat
+import numpy as np
+
+from ..basesortingextractorinterface import BaseSortingExtractorInterface
+
+
+class CellExplorerSortingInterface(BaseSortingExtractorInterface):
+    """Primary data interface class for converting Cell Explorer spiking data."""
+
+    SX = se.CellExplorerSortingExtractor
+
+    def get_metadata(self):
+        """Auto-populates spiking unit metadata."""
+        session_path = Path(self.source_data['spikes_matfile_path']).parent
+        session_id = session_path.stem
+        # TODO: add condition for retrieving ecephys metadata if no recoring or lfp are included in conversion
+
+        unit_properties = []
+        cell_filepath = session_path / f"{session_id}.spikes.cellinfo.mat"
+        if cell_filepath.is_file():
+            cell_info = loadmat(cell_filepath).get('spikes', np.empty(0))
+            cell_info_fields = cell_info.dtype.names
+            if 'cluID' in cell_info_fields:
+                unit_properties.append(
+                    dict(
+                        name="shank_id",
+                        description="0-indexed id of cluster identified from the shank.",
+                        # - 2 b/c the 0 and 1 IDs from each shank have been removed
+                        data=[int(x - 2) for x in cell_info['cluID'][0][0][0]]
+                    )
+                )
+            if 'shankID' in cell_info_fields:
+                unit_properties.append(
+                    dict(
+                        name="electrode_group",
+                        description="The electrode group that each unit was identified by.",
+                        data=[f"shank{x}" for x in cell_info['shankID'][0][0][0]]
+                    )
+                )
+            if 'region' in cell_info_fields:
+                unit_properties.append(
+                    dict(
+                        name="region",
+                        description="Brain region where each unit was detected.",
+                        data=[str(x[0]) for x in cell_info['region'][0][0][0]]
+                    )
+                )
+
+        celltype_mapping = {'pE': "excitatory", 'pI': "inhibitory"}
+        celltype_filepath = session_path / f"{session_id}.CellClass.cellinfo.mat"
+        if celltype_filepath.is_file():
+            celltype_info = loadmat(celltype_filepath).get('CellClass', np.empty(0))
+            if 'label' in celltype_info.dtype.names:
+                unit_properties.append(
+                    dict(
+                        name="cell_type",
+                        description="Type of cell this has been classified as.",
+                        data=[str(celltype_mapping[x[0]]) for x in celltype_info['label'][0][0][0]]
+                    )
+                )
+
+        return dict(UnitProperties=unit_properties)
```

Comment:
Wouldn't this be

```python
return dict(
    Ecephys=dict(
        UnitProperties=unit_properties
    )
```
?

Reply:
The `BaseSortingInterface` is currently setup to pull metadata from a first-level field `UnitProperties`.

This doesn't appear to have been added to metadata schema documentation so I'll look into that; however, these are also all completely custom/optional so it's not a validation/enforcement thing.

We can discuss it further, but I think it's conceivable that there is spiking activity inferred or included outside the context of `Ecephys`, so I would personally want to keep it at that first level.

Reply:
Would it be possible that a SortingInterface would add metadata to other fields of the metadata dict? e.g. session or subject info?

Reply:
I can't think of any use case like that from everything I've seen, except the open question of how best to allow non-recording interfaces to write/call the writing of `Ecephys` info into the metadata dictionary in the case where there is no actual `RecordingInterface` (some of those changes are proposed [on this separate PR](https://github.com/catalystneuro/nwb-conversion-tools/pull/102)) but it might be better to open this as a discussion on Slack to see if other people ever have odd use cases like that. I'll get that convo started in a bit

Reply:
I think that's a great use-case. Also 

* The `sessionName.spikes.cellinfo.mat` file has a `sessionName` attribute that could be used as an `NWBFile.session_id`
* The Vermin data contains subject id

I'm sure there are more. I don't really see a downside in passing back an entire metadata dict.